### PR TITLE
[IMP] base: faster view validation

### PR DESCRIPTION
**Pull request by rco-odoo**
This proposes a split of post-processing methods to make easier optimizations specific to each use case (post-processing and validation).  It also proposes other changes that make the code more readable.

With all those changes, the time spent in method `_check_xml()` to install a database from scratch with
module `sale_management` went from 5.145s to 2.243s (**-56%**), and the number of queries made in the method went from 3779 to 1614 (**-57%**).

### CODE REVIEW

File: odoo/addons/base/models/ir_ui_view.py [hunk 1]
```diff
@@ -41,6 +41,26 @@
...
+    # most (~95%) elements are 1/True/0/False
+    if expr == '1':
+        return 1
+    if expr == 'True':
+        return True
+    if expr == '0':
+        return 0
+    if expr == 'False':
+        return False
+    return safe_eval.safe_eval(expr, globals_dict)
```
Comment (fmdl) [inline odoo/addons/base/models/ir_ui_view.py:55]: maybe add this in safe_eval()

Comment (rco-odoo) [reply odoo/addons/base/models/ir_ui_view.py:55]: The other places where we use `safe_eval()` generally don't expect those values.  Those values are very common in views, but not at all in server actions or `ir.rule` domains, for instance.  That's why I think this optimization is more specific to views, and should remain specific.

**Review by Julien00859 — COMMENTED** [review]
This is great!

The code is much more understandable and idiomatique now than it was in say v12. My primary concern is about the `_combine` method that is kinda bloated now plus a few other questions. 

The split in many commits did help me understand every moving part which did ease my review on the final outcome. :+1:

File: odoo/addons/base/models/ir_ui_view.py [hunk 2]
```diff
@@ -700,35 +700,47 @@ def _check_view_access(self):
...
+        """
+        error_context = {
+            'view': self,
+            'name': getattr(self, 'name', None),
+            'xmlid': self.env.context.get('install_xmlid') or self.xml_id,
+            'view.model': self.model,
+            'view.parent': self.inherit_id,
+            'file': self.env.context.get('install_filename'),
+            'line': node.sourceline if node is not None else 1,
+        }
```
[the remark this answers is not in the document]
Comment (Julien00859) [reply odoo/addons/base/models/ir_ui_view.py:757]: We don't live in France, we don't speak french.

File: odoo/addons/base/models/ir_ui_view.py [hunk 3]
```diff
@@ -1213,28 +1189,151 @@ def _postprocess_tag_tree(self, node, name_manager, node_info):
...
+        return name_manager
+
     #------------------------------------------------------
     # Node validator
     #------------------------------------------------------
+    def _validate_tag_form(self, node, name_manager, node_info):
+        pass
+
+    def _validate_tag_tree(self, node, name_manager, node_info):
+        self._validate_tag_form(node, name_manager, node_info)
```
[the remark this answers is not in the document]
Comment (Julien00859) [reply odoo/addons/base/models/ir_ui_view.py:1288]: Override ?

[the remark this answers is not in the document]
Comment (rco-odoo) [reply odoo/addons/base/models/ir_ui_view.py]: An editable tree view behaves mostly like a form view, therefore we can reuse form view validation.

**Comment by rco-odoo** [comment]
@robodoo rebase-ff

**Comment by rco-odoo** [comment]
@robodoo override=ci/security r+
About override: a call to `safe_eval` was simply moved to a proxy function.

**Comment by rco-odoo** [comment]
@robodoo override=ci/security
a call to safe_eval was simply moved to a proxy function

**Comment by rco-odoo** [comment]
@robodoo r+

**Pull request closed without merging**